### DISCUSSION TITLE
Use semantic versioning

### DISCUSSION THREAD

**Discussion by timothyqiu** [godotengine/godot-proposals#8938] [General Discussions]
Currently, Godot loosely follows semantic versioning. Minor versions are allowed to break existing projects. This is acceptable in 3.x era because a minor version lasts for about a year.

However, 4.x has a faster pace, resulting in much more frequent breakages. This is especially frustrating for tutorial makers. The tutorials are very likely to be invalidated every few months. For example, `change_scene_to_file()` is such a fundamental & big change.

Please keep things compatible between minor versions.

**Comment by Mickeon** [comment]
Godot does use semantic versioning. Well, at least it is supposed to. 🙃 But of course, I gather this is a call for the maintainers to be more careful for what they merge in minor versions.
